src/Ocelot/Configuration/File/FileUpstreamHeaderRoutingOptions.cs

```diff
@@ -0,0 +1,17 @@
+using System.Collections.Generic;
+
+namespace Ocelot.Configuration.File
+{
+    public class FileUpstreamHeaderRoutingOptions
+    {
+        public FileUpstreamHeaderRoutingOptions()
+        {
+            Headers = new();
+            TriggerOn = string.Empty;
+        }
+
+        public Dictionary<string, List<string>> Headers { get; set; }
```

Comment:
```suggestion
        public IDictionary<string, IList<string>> Headers { get; set; }
```

If "List" is not mandatory, prefer string[] or IReadCollection<string> (les memory consumption)